suite-native/discovery/src/discoveryMiddleware.ts

```diff
@@ -55,6 +65,29 @@ export const prepareDiscoveryMiddleware = createMiddlewareWithExtraDeps(
             );
         }
 
+        // this is for further continual rerunning discovery for various other reasons
+        if (
+            selectDeviceThunk.fulfilled.match(action) ||
+            setEnabledDiscoveryNetworkSymbols.match(action) ||
+            toggleEnabledDiscoveryNetworkSymbols.match(action) ||
+            accountsActions.changeAccountVisibility.match(action) ||
+            deviceActions.updateSelectedDevice.match(action)
```

Comment:
Is this necessary? `updateSelectedDevice` happens on change on the device 🤔

Reply:
This is here to check in case the newly selected device has not finished discovery.

Reply:
and sometimes the device does not have state after being selected

Reply:
but `updateSelectedDevice` is not action of selecting new device - it is action of synchronization any change on the currently selected device from devices list to selectedDevice

Reply:
8f5b887272